src/coloring.rs

```diff
@@ -26,13 +26,36 @@ use petgraph::visit::NodeCount;
 
 use rayon::prelude::*;
 
-/// Color a PyGraph using a largest_first strategy greedy graph coloring.
+/// Color a :class:`~.PyGraph` object using a greedy graph coloring algorithm
+///
+/// This function uses a `largest-first` strategy as described in [1]_ and colors
+/// the nodes with higher degree first. As the coloring problem is NP-hard
+/// this is a heuristic algorithm and may not return an optimal solution.
```

Comment:
How about putting the warning into a `..note::` to make it better visible? 🙂 We should also add it to `SparsePauliOp.group_commuting` in Qiskit Terra, which might be where users are coming from (I can add it there 🙂)

Reply:
Done in: https://github.com/Qiskit/rustworkx/pull/857/commits/9243086e21e2d9556a40aa17dcd7d0efacfe4595